api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java

```diff
@@ -269,11 +269,16 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
 
     @Override
     public <T> Boolean in(BoundReference<T> ref, Set<T> literalSet) {
-      return ROWS_MIGHT_MATCH;
+      // in(col, {X, Y}) => eq(col, x) OR eq(col, x)
```

Comment:
This should copy the logic from `eq` instead of calling it directly so that the lower and upper bounds are accessed and converted just once.

Reply:
Done.